### ISSUE TITLE
Scheduling Algorithm

### ISSUE THREAD

**Issue by kvverti** [Apexal/late#80]
Given a user's schedule, split assignments into blocks and assign the blocks to free time. 

* [ ] Split assignments into appropriately sized blocks
* [ ] Order blocks based on priority, due date, common sense
* [ ] Assign blocks to free time

**Comment by kvverti** [comment]
We discussed the algorithm in the last meeting and settled on an initial picture. To get something off the ground, we will use equal-sized blocks, and use a weighted shuffle to order them (placing higher priority blocks earlier in general). Blocks are placed in order in the free time.

**Comment by Apexal** [comment]
The user can now set time preferences (earliest and latest) as well as block off **unavailable** time in their schedule. This means we can get a pretty good idea of a user's study/work schedule by taking these both into account.

**Comment by Apexal** [comment]
bumping

**Comment by aidan-lane** [comment]
Is this still being worked on? I think this would be cool to work on.

**Comment by Shadowsoftime99** [comment]
Yes! I have some very rough drafting of things on my laptop from last spring, but I'd like to talk with Frank (and maybe Thalia if we can get ahold of them) to understand the previously proposed algorithm. I also have my own ideas.

I have missed a few meetings before today, but you should be able to find me (Tyler) at small group usually.
